### ISSUE TITLE
Support for sorting inline NSG rules in plan output

### ISSUE THREAD

**Issue by mwsutherland** [hashicorp/terraform-provider-azurerm#21794]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Using inline rules for NSGs is great for making sure new, unmanaged rules don't creep in, but it is a nightmare for evaluating plans because the plan shows the entire rule set before and after. Asking for individual rules to be diffed seems like too much to ask, but could the rules be sorted in the list? Sorted by priority first for incoming and then by outgoing would be great but, at this point, I don't care what they are sorted by as long the before and after are sorted the same way.

### New or Affected Resource(s)/Data Source(s)

azurerm_network_security_group

**Comment by magodo** [comment]
@mwsutherland The nested NSRs are defined as a `TypeSet`, which indicates the order of the NSRs doesn't matter (won't cause plan diff). Currently, the underlying SDK doesn't allow the developer to keep the order after setting the values to data set as the `TypeSet` [internally keep the values in a Go map](https://github.com/hashicorp/terraform-provider-azurerm/blob/d70e6ea709f2c377ddc44620e66ed52a23d50dc1/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/set.go#L51-L56), which is out of order.

**Comment by mwsutherland** [comment]
I was afraid that, since it wasn't already being done, there was a reason for it, but I had to ask. Thanks for the explanation!

**Comment by mwsutherland** [comment]
Is the output of the diff (the only place the order matters) handled at a level so generic that it doesn't even know what the data it is displaying represents?

**Comment by mwsutherland** [comment]
I just made a big discovery that makes this request moot. I'm going to describe what I found in case it helps someone in the future.

It seems that there are 2 levels of differences. There are differences that make TF say "hey, we need to change this resource" and there are differences that don't require a change, but will be displayed in the plan output. 

It turns out that the reason I would see all the rules in the plan when there was a change in any rule was because of this 2nd type. So, when one rule has a substantial change, all the rules would show up, making it hard to see that one change. Diffing the plan output, I found that if you have a destination_address_prefixes or source_address_prefixes property, the plan would show that as well as empty destination_address_prefix or source_address_prefix properties. If I add those empty properties to my tf file, poof, those rules disappear from the plan output leaving just the change I expect!
